platform/components/insights/clusters/clusters-sheet.tsx

```diff
@@ -81,22 +83,86 @@ const RunClusters = ({
   const hobby = orgMetadata?.plan === "hobby";
 
   const [loading, setLoading] = React.useState(false);
-  const [scope, setScope] = React.useState("messages");
-  const [instruction, setInstruction] = React.useState("");
+
+  if (!project_id) {
+    return <></>;
+  }
+
+  function setSheetOpenWrapper(value: boolean) {
+    // Reset the dataFilters when the sheet is closed
+    if (!value) {
+      const currentDataFilters = dataFilters;
+      delete currentDataFilters.clustering_id;
+      delete currentDataFilters.clusters_ids;
+      setDataFilters(dataFilters);
+    }
+    setSheetOpen(value);
+  }
+
+  const FormSchema = z.object({
+    scope: z.enum(["messages", "sessions"]),
+    instruction: z
+      .string({
+        required_error: "Please enter an instruction",
+      })
+      .max(32, "Instruction must be at most 32 characters long"),
+    nb_clusters: z.number({
+      required_error: "Please enter the number of clusters",
+    }),
+  });
+
+  console.log("totalNbTasks: ", totalNbTasks);
+  const form = useForm<z.infer<typeof FormSchema>>({
+    resolver: zodResolver(FormSchema),
+    defaultValues: {
+      scope: "messages",
+      instruction: "user intent",
+      // Note: don't set the nb_clusters default value here, since it's updated dynamically using an API call
+    },
+  });
 
   useEffect(() => {
     if (totalNbTasks) {
       setClusteringCost(totalNbTasks * 2);
     }
-  }, [totalNbSessions, totalNbTasks, scope]);
+    // Update the default number of clusters when the total number of tasks changes
+    if (form.getValues("scope") === "messages") {
+      if (!totalNbTasks) {
+        totalNbTasks = 0;
+      }
+      setNbElements(totalNbTasks);
 
-  if (!project_id) {
-    return <></>;
-  }
+      if (Math.floor(totalNbTasks / 100) >= 5) {
+        form.setValue("nb_clusters", Math.floor(totalNbTasks / 100));
+      } else {
+        form.setValue("nb_clusters", 5);
+      }
+    }
+    if (form.getValues("scope") === "sessions") {
+      if (!totalNbSessions) {
+        totalNbSessions = 0;
+      }
+      setNbElements(totalNbSessions);
+      if (Math.floor(totalNbSessions / 100) >= 5) {
```

Comment:
please use a variable instead of writing the same expression twice

Reply:
Done, new variable defaultNbClusters